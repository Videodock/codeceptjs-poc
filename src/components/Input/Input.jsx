import React from 'react';
import { makeStyles } from '@material-ui/styles';
import * as PropTypes from 'prop-types';
import TextField from '@material-ui/core/TextField';
import Typography from '@material-ui/core/Typography';
import styles from './Input.styles';

const useStyles = makeStyles(styles);

export default function Input ({ handleChange, handleBlur, value, label, name, endAdornment, children, ...rest }) {
  const classes = useStyles();

  return (
    <div className={classes.root}>
      <Typography className={classes.inputHeader} variant="body1">
        {label}
      </Typography>
      <TextField
        onChange={handleChange}
        onBlur={handleBlur}
        name={name}
        value={value}
        variant="filled"
        InputProps={{
          disableUnderline: true,
          endAdornment,
        }}
        fullWidth
        {...rest}
      >{children}
      </TextField>
    </div>
  );
}

Input.propTypes = {
  handleBlur  : PropTypes.func,
  handleChange: PropTypes.func.isRequired,
  value       : (PropTypes.number || PropTypes.string).isRequired,
  label       : PropTypes.string,
};
