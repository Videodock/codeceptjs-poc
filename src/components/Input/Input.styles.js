import { createStyles } from '@material-ui/styles';

export default ({ spacing }) => createStyles({
  root: {
    '& .MuiTextField-root': {
      borderRadius: '25px',
    },
    marginBottom: spacing(2.5),
  },
});
