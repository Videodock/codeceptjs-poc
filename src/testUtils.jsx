import React from 'react';
import { Provider } from 'react-redux';
import { render } from '@testing-library/react';
import { ThemeProvider } from '@material-ui/styles';
import { I18nextProvider } from 'react-i18next';
import { ApolloProvider } from '@apollo/client';
import { BrowserRouter as Router } from 'react-router-dom';
import 'url-search-params-polyfill';
import i18n from '../src/i18n';
import client from './graphql';
import theme from './theme';
import store from './redux';

const wrapper = ({ children }) => (
  <ApolloProvider client={client}>
    <Provider store={store}>
      <ThemeProvider theme={theme}>
        <I18nextProvider i18n={i18n}>
          <Router>
            {children}
          </Router>
        </I18nextProvider>
      </ThemeProvider>
    </Provider>
  </ApolloProvider>
);

const customRender = (ui, options) => render(ui, { wrapper, ...options });

export { customRender as render };
