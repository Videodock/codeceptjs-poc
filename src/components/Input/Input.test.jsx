import React from 'react';
import { render } from '../../testUtils';
import Input from './Input';

describe('<Input />', () => {
  test('renders and matches snapshot', () => {
    const { container } = render(<Input />);

    expect(container.firstChild).toMatchSnapshot();
  });

  // add your unit tests here
});
