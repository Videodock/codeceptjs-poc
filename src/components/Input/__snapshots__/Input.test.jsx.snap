// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Input /> renders and matches snapshot 1`] = `
<div
  class="makeStyles-root-1"
>
  <p
    class="MuiTypography-root MuiTypography-body1"
  />
  <div
    class="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
  >
    <div
      class="MuiInputBase-root MuiFilledInput-root MuiInputBase-fullWidth MuiInputBase-formControl"
    >
      <input
        aria-invalid="false"
        class="MuiInputBase-input MuiFilledInput-input"
        type="text"
        value=""
      />
    </div>
  </div>
</div>
`;
